### ISSUE TITLE
Add function to get position of named capture

### ISSUE THREAD

**Issue by dexamphetamine** [rust-lang/regex#276]
It would be really, really convenient to get the position of a match returned by `re.captures("something").name("capture_group_name")`.

I made a first attempt at implementing it based on what's available, but it's both buggy and inefficient...

``` rust
extern crate regex;
use regex::{Regex, RegexSet, Captures};


fn get_named_capture_index(caps: &Captures, cap_name: &str) -> Option<usize> {
  let name_string = caps.name(cap_name);
  let mut res: Option<usize> = None;
  if let Some(name_string) = name_string {
    for (cap_ind, _) in caps.iter_pos().enumerate() {
      res = match caps.at(cap_ind) {
        Some(ind_string) => {
          if ind_string == name_string {
            Some(cap_ind)
          } else {
            None
          }
        },
        None => None
      };
    }
  }
  res
}

fn get_named_capture_position(caps: &Captures, cap_name: &str) ->
  Option<(usize, usize)> {
    let cap_ind = get_named_capture_index(caps, cap_name);
    if let Some(cap_ind) = cap_ind {
      caps.pos(cap_ind)
    } else {
      None
    }
}
```

**Comment by BurntSushi** [comment]
This should be fixed once `1.0` is released. Specifically, the `name` (and `at`/`get`) methods now return a `Match`, which includes both the position information and the matched substring. See here: https://github.com/rust-lang-nursery/regex/blob/rfc/src/re_unicode.rs#L926

A work around you could use today is to build a map of capture name to index for the entire index using [`capture_names`](https://doc.rust-lang.org/regex/regex/struct.Regex.html#method.capture_names), which returns an iterator of `Option<&str>`. You'd only have to do this once for each regex, and you could look up the name in that map to  get the capture index, and then you could use your `get_named_capture_position` function.

**Comment by dexamphetamine** [comment]
Thanks for the workaround! If anyone else happens on this ticket before 1.0, here is what I wrote following your advice: 

```
fn get_named_capture_index(re: &Regex, cap_name: &str) -> Option<usize> {
  re.capture_names().position(|name| {
    name == Some(cap_name)
  })
}

fn get_named_capture_position(re: &Regex, caps: &Captures, cap_name: &str) ->
  Option<(usize, usize)> {
    let cap_ind = get_named_capture_index(re, cap_name);
    if let Some(cap_ind) = cap_ind {
      caps.pos(cap_ind)
    } else {
      None
    }
}```
